scipy/spatial/_polygon_surface_area.py

```diff
@@ -0,0 +1,153 @@
+from . import _surface_area
+from six.moves import xrange
+
+#
+# Copyright (C)  James Nichols and Tyler Reddy
+# Contributions from Tyler Reddy are owned by LANL
+# and respect the scipy distribution license
+#
+# Distributed under the same BSD license as Scipy.
+
+def convert_cartesian_array_to_spherical_array(coord_array,angle_measure='radians'):
+    '''Take shape (N,3) cartesian coord_array and return an array of the same shape in spherical polar form (r, theta, phi). Based on StackOverflow response: http://stackoverflow.com/a/4116899
+    use radians for the angles by default, degrees if angle_measure == 'degrees' '''
+    spherical_coord_array = np.zeros(coord_array.shape)
+    xy = coord_array[...,0]**2 + coord_array[...,1]**2
+    spherical_coord_array[...,0] = np.sqrt(xy + coord_array[...,2]**2)
+    spherical_coord_array[...,1] = np.arctan2(coord_array[...,1], coord_array[...,0])
+    spherical_coord_array[...,2] = np.arccos(coord_array[...,2] / spherical_coord_array[...,0])
+    if angle_measure == 'degrees':
+        spherical_coord_array[...,1] = np.degrees(spherical_coord_array[...,1])
+        spherical_coord_array[...,2] = np.degrees(spherical_coord_array[...,2])
+    return spherical_coord_array
+
+def convert_spherical_array_to_cartesian_array(spherical_coord_array,angle_measure='radians'):
+    '''Take shape (N,3) spherical_coord_array (r,theta,phi) and return an array of the same shape in cartesian coordinate form (x,y,z). Based on the equations provided at: http://en.wikipedia.org/wiki/List_of_common_coordinate_transformations#From_spherical_coordinates
+    use radians for the angles by default, degrees if angle_measure == 'degrees' '''
+    cartesian_coord_array = np.zeros(spherical_coord_array.shape)
+    #convert to radians if degrees are used in input (prior to Cartesian conversion process)
+    if angle_measure == 'degrees':
+        spherical_coord_array[...,1] = np.deg2rad(spherical_coord_array[...,1])
+        spherical_coord_array[...,2] = np.deg2rad(spherical_coord_array[...,2])
+    #now the conversion to Cartesian coords
+    cartesian_coord_array[...,0] = spherical_coord_array[...,0] * np.cos(spherical_coord_array[...,1]) * np.sin(spherical_coord_array[...,2])
+    cartesian_coord_array[...,1] = spherical_coord_array[...,0] * np.sin(spherical_coord_array[...,1]) * np.sin(spherical_coord_array[...,2])
+    cartesian_coord_array[...,2] = spherical_coord_array[...,0] * np.cos(spherical_coord_array[...,2])
+    return cartesian_coord_array
+
+def _vertex_index_strider(index, num_vertices):
+    # handle the wrapping / iteration over
+    # polygon vertices in either CW or CCW
+    # sort order
+    forward_index = index + 1
+    backward_index = index - 1
+    if forward_index >= num_vertices:
+        forward_index = 0
+    return forward_index, backward_index
+
+def _planar_polygon_area(vertices):
+    num_vertices = vertices.shape[0]
+    area_sum = 0
+    for i in xrange(num_vertices):
```

Comment:
Have you profiled this? How long does this code take? It might be worth moving to Cython.

Reply:
Yeah, see the checklist above -- I think I'll just switch to Cython / C-only moving forward. I tried to develop in Python / Cython side-by-side (to compare the speed of the two) but the dual implementation approach during development eventually got neglected and I mostly used Python for convenience.

That said, we're probably still quite far from "production ready" to worry about performance too much.